contrib/survey_cake/udfs/survey_cake_csv_uploader.py

```diff
@@ -0,0 +1,136 @@
+
+        self.facttable_filepath = (
+            self.filename.parent / f"{self.filename.stem}_facttable.csv"
+        )
+        self.dimension_table_filepath = (
+            self.filename.parent / f"{self.filename.stem}_dimension.csv"
+        )
+
+    @property
+    def year(self):
+        return self._year
+
+    @property
+    def bigquery_project(self):
+        return os.getenv("BIGQUERY_PROJECT")
+
+    def transform(self):
+        def _export_facttable(header_of_fact_table):
+            with open(self.facttable_filepath, "w") as target:
+                writer = csv.writer(target)
+                writer.writerow(header_of_fact_table)
+                for row in rows_of_fact_table:
+                    row_with_year = row + (self.year,)
+                    writer.writerow(row_with_year)
+
+        def _export_dimension_table(question_id_dimension_table):
+            with open(self.dimension_table_filepath, "w") as target:
+                writer = csv.writer(target)
+                writer.writerow(("question_id", "question", "year"))
+                for question_id, question in question_id_dimension_table.items():
+                    writer.writerow((question_id, question, self.year))
+
+        with open(Path(self.filename), "r", encoding="utf-8-sig") as csvfile:
+            rows = csv.reader(csvfile)
+            # skip header
+            header = next(iter(rows))
+            question_id_dimension_table = self._generate_question_id_dimension_table(
+                header
+            )
+            question_ids = sorted(question_id_dimension_table.keys())
+            header_of_fact_table = ("ip", "question_id", "answer", "year")
+            rows_of_fact_table = self._transform_raw_data_to_fact_table_format(
+                rows, question_id_dimension_table, question_ids
+            )
+
+        _export_facttable(header_of_fact_table)
+        _export_dimension_table(question_id_dimension_table)
+
+    def upload(
+        self,
+        facttable_or_dimension_table,
+        data_layer,
+        data_domain,
+        primary_key,
+        time_dimension,
+    ):
+        if facttable_or_dimension_table == "fact":
+            self._upload_2_bigquery(
+                self.facttable_filepath,
+                f"{self.bigquery_project}.{data_layer}.{data_layer}_{data_domain}_{primary_key}_{time_dimension}",
+            )
+        elif facttable_or_dimension_table == "dim":
+            self._upload_2_bigquery(
+                self.dimension_table_filepath,
+                f"{self.bigquery_project}.{data_layer}.{data_layer}_{data_domain}_{primary_key}_{time_dimension}",
+            )
+
+    def _upload_2_bigquery(self, file_path, table_id):
+        job_config = bigquery.LoadJobConfig(
+            source_format=bigquery.SourceFormat.CSV,
+            skip_leading_rows=1,
+            autodetect=True,
+            allow_quoted_newlines=True,
+            write_disposition="WRITE_APPEND",
```

Comment:
caveat: use `WRITE_APPEND` in place of `WRITE_TRUNCATE`

otherwise it would overwrite all the existing data in BigQuery!